packages/mendel-middleware/index.js

```diff
@@ -29,6 +30,38 @@ function MendelMiddleware(opts) {
             return getPath({ bundle: bundle, hash: tree.hash });
         };
 
+        req.mendel.getExposed = function(variations) {
+            if (!variations) {
+                variations = 'base';
+            }
+
+            if (exposedBundleVariations[variations]) {
+                return exposedBundleVariations[variations];
+            }
+
+            var exposedBundleVariation = Object.keys(trees.bundles).reduce(function(acc, bundle) {
+                var vTree = trees.findTreeForVariations(bundle, variations);
```

Comment:
Agreed this could be performance bottleneck for the first request of a specific variation. After the first request, it should be served form `exposedBundleVariations` object as a cache.